### ISSUE TITLE
iam-group add "unused" filter which can reuse "has-users: false" filter

### ISSUE THREAD

**Issue by nitrocode** [cloud-custodian/cloud-custodian#5464]
**Is your feature request related to a problem? Please describe.**
Consistency. Here are some "unused" filters that exist

```
aws.ami.filters.unused
aws.ebs-snapshot.filters.unused
aws.iam-policy.filters.unused
aws.iam-profile.filters.unused
aws.iam-role.filters.unused
aws.launch-config.filters.unused
aws.security-group.filters.unused
```

but for `iam-group`, it's called `aws.iam-group.filters.has-users`

**Describe the solution you'd like**
Instead, or on top, of this

```
    filters:
      - type: has-users
        value: False
```

also allow this to keep it consistent with other resources

```
    filters:
      - unused
```

**Describe alternatives you've considered**

**Additional context**

**Comment by kapilt** [comment]
yes, the iam resources have some deprecated syntax/style around its filters, till we have a deprecation framework I'm loathe to multiple ways to express the same filter in different ways without a way to cut down on the load by giving guidance as to best/current practice.

although in this particular case perhaps its a useful alias as they are different semantics for a user reasoning about a domain.
